### DISCUSSION TITLE
Consider passing `Actor#thing` to group membership blocks?

### DISCUSSION THREAD

**Discussion by grncdr** [jnunemaker/flipper#569] [Q&A] [answered]
So I recently tried flipper and had the exact same experience as #475. I started with an `actor.is_a?(Something) && actor.does_something?` check, which was failing because `actor` is a `Flipper::Types::Actor` instance. Is there any reason the block should receive the `Actor` wrapper and not the `thing` it wraps?

**Comment by jnunemaker** [comment, marked as answer]
There is a reason, but it was long ago and I don't recall off hand. I'd dig in and figure out why, but we are actually working on some amazing stuff that I think will replace groups so I'm not inclined to dig in right now. 

Any reason not to use respond_to?(:some_method) to go the quacks like a duck route instead of the is a duck route? Just curious.

**Reply by grncdr** [reply]
The main reason is that in my particular project, there are two types of users (with distinct ActiveRecord models), and I'd like to use Flipper for toggling features on them both. The `is_a?` checks are already somewhat idiomatic in this project (e.g. for ActionCable) so it was just the natural way to start expressing groups.

>  we are actually working on some amazing stuff that I think will replace groups so I'm not inclined to dig in right now.

I wouldn't worry too much about this then, but depending on your schedule, it _might_ be worth mentioning the `is_a?` caveat in the docs. The `Flipper.register` API and the documentation strongly imply that the object you pass to `enabled?` is the one that will be later received by the block.
